app/core/file_processor/langchain_loader.py

```diff
@@ -27,36 +26,64 @@ def process_file(
         with custom handling for its format and contents.
         Implementations should try to fill as much additional information like links, media etc.
         label, when provided, should apply to all documents in the o/p list"""
+        name = kwargs.get("name", "unknown")
         if file_type in [schema.FileType.TEXT, schema.FileType.MD]:
-            name = kwargs.get("name", None)
             metadata = kwargs.get("metadata", {})
-            output_list = self.process_file_text(
-                file=file, label=label, name=name, metadata=metadata
-            )
+            texts = self.text_loader(file_path = file_path)
+            output_list = self.process_texts(
+                texts = texts,
+                label=label,
+                name=name,
+                metadata=metadata
+                )
+        elif file_type == schema.FileType.PDF:
+            metadata = kwargs.get("metadata", {})
+            texts = self.pdf_loader(file_path = file_path)
+            output_list = self.process_texts(
+                texts = texts,
+                label=label,
+                name=name,
+                metadata=metadata
+                )
         elif file_type == schema.FileType.CSV:
             args = {}
             col_delimiter = kwargs.get("col_delimiter")
             if col_delimiter is not None:
                 args["col_delimiter"] = col_delimiter
-            output_list = self.process_file_csv(file=file, **args)
+            output_list = self.process_file_csv(file=file_path, **args)
         else:
             raise GenericException("This file type is not supported (yet)!")
         return output_list
 
-    def process_file_text(
-        self, file: TextIOWrapper, label: str, name: str = None, metadata: dict = None
+
+    def text_loader(self, file_path: str) -> List[schema.Document]:
+        """Uses langchain's TextLoader to load text contents from file"""
+        loader = TextLoader(file_path)
+        texts = loader.load()
+        return texts
+
+
+    def pdf_loader(self, file_path: str) -> List[schema.Document]:
+        """Uses langchain's UnstructuredPDFLoader to load text contents from file"""
+        loader = UnstructuredPDFLoader(file_path)
+        texts = loader.load()
+        return texts
+
+
+    def process_texts(
+        self,
+        texts: List[schema.Document],
+        label:str,
+        name: str = None,
+        metadata: dict = None,
     ) -> List[schema.Document]:
         """Uses langchain's TokenTextSplitter to convert text contents into document format"""
         output_list = []
-        loader = TextLoader(file)
-        texts = loader.load()
         text_splitter = TokenTextSplitter(chunk_size=1000, chunk_overlap=50)
         text_splits = text_splitter.split_documents(texts)
 
         if not label:
             label = "open-access"
-        if name is None or name.strip() == "":
-            name = file.name
```

Comment:
Dont we want this default naming convention with name of uploaded file? Is see it could be None... Or is it handled somewhere?

Reply:
Oh you are setting it to unknown. Is n't it better to keep the filename?

Reply:
It would be better to keep the filename, but here `file` is the filename string - it's not a file object that has a `.name` attribute. So we don't have access to that name at this point.

Reply:
We could default it to the filename string - I'll do that.

Reply:
I see this is how you pass the value from router `file_path=f"{UPLOAD_PATH}{file_obj.filename}"`. We are keeping the original file name in the file_path. It could be obtained using a split() may be inside the fileprocessor module

Reply:
OK yep, I've done that now.